liana-gui/src/export.rs

```diff
@@ -0,0 +1,386 @@
+        Daemon, DaemonBackend, DaemonError,
+    },
+    lianalite::client::backend::api::DEFAULT_LIMIT,
+};
+
+macro_rules! send_error {
+    ($sender:ident, $error:ident) => {
+        if let Err(e) = $sender.send(ExportProgress::Error(Error::$error)) {
+            tracing::error!("ExportState::start() fail to send msg: {}", e);
+        }
+    };
+    ($sender:ident, $error:expr) => {
+        if let Err(e) = $sender.send(ExportProgress::Error($error)) {
+            tracing::error!("ExportState::start() fail to send msg: {}", e);
+        }
+    };
+}
+
+macro_rules! send_progress {
+    ($sender:ident, $progress:ident) => {
+        if let Err(e) = $sender.send(ExportProgress::$progress) {
+            tracing::error!("ExportState::start() fail to send msg: {}", e);
+        }
+    };
+    ($sender:ident, $progress:ident($val:expr)) => {
+        if let Err(e) = $sender.send(ExportProgress::$progress($val)) {
+            tracing::error!("ExportState::start() fail to send msg: {}", e);
+        }
+    };
+}
+
+#[derive(Debug, Clone)]
+pub enum ExportMessage {
+    Open,
+    ExportProgress(ExportProgress),
+    TimedOut,
+    UserStop,
+    Path(Option<PathBuf>),
+    Close,
+}
+
+impl From<ExportMessage> for view::Message {
+    fn from(value: ExportMessage) -> Self {
+        Self::Export(value)
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum ExportState {
+    Init,
+    ChoosePath,
+    Path(PathBuf),
+    Started,
+    Progress(f32),
+    TimedOut,
+    Aborted,
+    Ended,
+    Closed,
+}
+
+#[derive(Debug, Clone)]
+pub enum Error {
+    Io(String),
+    HandleLost,
+    UnexpectedEnd,
+    JoinError(String),
+    ChannelLost,
+    NoParentDir,
+    Daemon(String),
+    TxTimeMissing,
+}
+
+impl From<JoinError> for Error {
+    fn from(value: JoinError) -> Self {
+        Error::JoinError(format!("{:?}", value))
+    }
+}
+
+impl From<std::io::Error> for Error {
+    fn from(value: std::io::Error) -> Self {
+        Error::Io(format!("{:?}", value))
+    }
+}
+
+impl From<DaemonError> for Error {
+    fn from(value: DaemonError) -> Self {
+        Error::Daemon(format!("{:?}", value))
+    }
+}
+
+#[derive(Debug)]
+pub enum Status {
+    Init,
+    Running,
+    Stopped,
+}
+
+#[derive(Debug, Clone)]
+pub enum ExportProgress {
+    Started(Arc<Mutex<JoinHandle<()>>>),
+    Progress(f32),
+    Ended,
+    Finished,
+    Error(Error),
+    None,
+}
+
+pub struct State {
+    pub receiver: Receiver<ExportProgress>,
+    pub sender: Option<Sender<ExportProgress>>,
+    pub handle: Option<Arc<Mutex<JoinHandle<()>>>>,
+    pub daemon: Arc<dyn Daemon + Sync + Send>,
+    pub path: Box<PathBuf>,
+}
+
+impl State {
+    pub fn new(daemon: Arc<dyn Daemon + Sync + Send>, path: Box<PathBuf>) -> Self {
+        let (sender, receiver) = channel();
+        State {
+            receiver,
+            sender: Some(sender),
+            handle: None,
+            daemon,
+            path,
+        }
+    }
+
+    pub async fn start(&mut self) {
```

Comment:
Is there a way to move the logic in this method to a module specific to the transactions export and keep only the general parts here?

Reply:
i've try this before but it was a real pain, i'd prefer keep this way for now (some weird issue about async)

Reply:
OK sure, we can come back to that the next time we need to add an export.